explorer/src/utils/fauna.ts

```diff
@@ -0,0 +1,111 @@
+import { AuthProvider } from '@/constants/session'
+import { Client, ExprArg, FaunaHttpErrorResponseContent, query as q } from 'faunadb'
+import type { SavedUser, User } from 'next-auth'
+import { headers } from 'next/headers'
+
+const client = new Client({
+  secret: process.env.FAUNA_DB_SECRET || '',
+  keepAlive: false,
+  queryTimeout: 2000,
+  timeout: 30,
+  http2SessionIdleTime: 1000,
+  domain: 'db.us.fauna.com',
+  scheme: 'https',
+})
+
+export const findUserByID = async (userID: string) => {
+  return await client
+    .query(q.Paginate(q.Match(q.Index('users_by_id'), userID)))
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    .then((response: any) => {
+      const resultsRefs = response.data
+      if (resultsRefs.length === 0) return null
+
+      const results = resultsRefs.map((ref: ExprArg) => {
+        return q.Get(ref)
+      })
+
+      return client.query(results) as Promise<{ ref: ExprArg; ts: number; data: SavedUser }[]>
+    })
+    .catch((error: FaunaHttpErrorResponseContent) => {
+      console.log('error', error)
+    })
+}
+
+export const findUserByAuthProviderId = async (provider: AuthProvider, id: string) => {
+  return await client
+    .query(q.Paginate(q.Match(q.Index(`users_by_${provider.toString()}_id`), id)))
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    .then((response: any) => {
+      const resultsRefs = response.data
+      if (resultsRefs.length === 0) return null
+
+      const results = resultsRefs.map((ref: ExprArg) => {
+        return q.Get(ref)
+      })
+
+      return client.query(results) as Promise<{ ref: ExprArg; ts: number; data: SavedUser }[]>
+    })
```

Comment:
This code is the exact same as in `findUersByID`. It would be cleaner to extract into its own function.